### DISCUSSION TITLE
Agent navigation on grid world

### DISCUSSION THREAD

**Discussion by bazakoskon** [mesa/mesa#1025] [Q&A]
Hi all and thanks for this awesome framework!
I was wondering if there are already any  implemented functions that help moving agents to efficiently navigate on gridworld (even with constraints, e.g obstacles). 
Algorithms such as shortest path between two cells etc.

Thanks ✌️

**Comment by tpike3** [comment]
@bazakoskon There is not but I am interested in building out the mesa ecosystem to make ABMs easier to grow. The idea to keep mesa core light is to store extensions. I have a nascent repo set up [projectmesabehaviors](https://github.com/projectmesabehaviors) intended to store such repos. (not officially part of project mesa) So happy to add a repo of grid world functions. (Full disclosure recently most of my time for this efforts has been used to develop  [data pipelines](https://github.com/projectmesadata) that allow for injection data into models and providing validation data

[Mesa-Geo](https://github.com/Corvince/mesa-geo) is also related to this if you consider geospatial navigation algorithms like A*

**Reply by mrak31** [reply]
This one(link) helped me alot as i was building an agent grid world with editable directional roads, spawn points, exits and walls like in a traffic intersection(working but not finished yet)

Hope this helps
https://github.com/Chadsr/MesaFireEvacuation
